src/java/com/zimbra/oauth/utilities/OAuthDataSource.java

```diff
@@ -129,7 +129,16 @@ public void syncDatasource(ZMailbox mailbox, OAuthInfo credentials) throws Servi
                 // ensure the specified storage folder exists
                 final String folderName = String
                     .format(OAuth2Constants.DEFAULT_OAUTH_FOLDER_TEMPLATE, username, client);
-                final String storageFolderId = ensureFolder(mailbox, folderName, View.contact);
+                View view = null;
+                if (dsType == DataSourceType.oauth2contact) {
```

Comment:
Don't map in this direction here. If you make the above change, just use `DataSourceType dsType = getDataSourceType(type);`.